test/proto.spec.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const proto = require('../src/protocol')
+
+describe('protocol', function () {
+  let msgObject = null
+  let message = null
+
+  before(() => {
+    msgObject = {
+      type: proto.CircuitRelay.Type.HOP,
+      srcPeer: {
+        id: 'QmSource',
+        addrs: [
+          '/p2p-circuit/ipfs/QmSource',
+          '/p2p-circuit/ipv4/0.0.0.0/9000/ipfs/QmSource',
+          'ipv4/0.0.0.0/9000/ipfs/QmSource'
+        ]
+      },
+      dstPeer: {
+        id: 'QmDest',
+        addrs: [
+          '/p2p-circuit/ipfs/QmDest',
+          '/p2p-circuit/ipv4/1.1.1.1/9000/ipfs/QmDest',
+          'ipv4/1.1.1.1/9000/ipfs/QmDest'
+        ]
+      }
+    }
+
+    let buff = proto.CircuitRelay.encode(msgObject)
+    message = proto.CircuitRelay.decode(buff)
+  })
+
+  it(`should source and dest`, () => {
+    expect(message.srcPeer).to.not.be.null()
```

Comment:
should be `deep.equal` check rather than `not.be.null`

Reply:
hmm, there is nothing to `deep.equal` to in this test, the message is the result of serializing an object literal to protobuf. I don't think it make sense to deep equal it to itself or de-serialize it one more time and then deep equal? Tho, I'm open to suggestions. 👌

Reply:
actually, I take that back! It makes perfect sense here ;-)